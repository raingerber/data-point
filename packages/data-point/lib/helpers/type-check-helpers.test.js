/* eslint-env jest */

const helpers = require('./type-check-helpers')
const typeCheckFunctions = require('./type-check-functions')

describe('type-check-helpers#normalizeTypeCheckSource', () => {
  test('It should return function reducer it matches any of the predefined types', () => {
    expect(helpers.normalizeTypeCheckSource('string')).toEqual(
      typeCheckFunctions.isString
    )
    expect(helpers.normalizeTypeCheckSource('number')).toEqual(
      typeCheckFunctions.isNumber
    )
  })

  test('It should return the reducer -as-is- if no match', () => {
    expect(helpers.normalizeTypeCheckSource('foo:bar')).toEqual('foo:bar')
  })

  test('It should process arrays that include some predefined types', () => {
    expect(
      helpers.normalizeTypeCheckSource(['string', 'foo:bar', 'number'])
    ).toEqual([
      typeCheckFunctions.isString,
      'foo:bar',
      typeCheckFunctions.isNumber
    ])
  })
})

describe('type-check-helpers#getTypeCheckSource', () => {
  test('should return defaultType when no specType is provided', () => {
    expect(helpers.getTypeCheckSource('hash', 'string')).toBe('string')
  })
  test('should return defaultType when specType === defaultType', () => {
    expect(helpers.getTypeCheckSource('hash', 'string', 'string')).toBe(
      'string'
    )
  })
  test('should throw error for invalid specType', () => {
    expect(() =>
      helpers.getTypeCheckSource('hash', 'string', 'number')
    ).toThrowErrorMatchingSnapshot()
  })
  test('should return array when specType is valid and !== defaultType', () => {
    const customTypeCheck = () => true
    expect(
      helpers.getTypeCheckSource('hash', 'string', customTypeCheck)
    ).toEqual(['string', customTypeCheck])
  })
})
