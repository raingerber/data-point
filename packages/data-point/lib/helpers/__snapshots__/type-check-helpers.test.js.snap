// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`type-check-helpers#getTypeCheckSourceWithDefault should throw error for invalid specType 1`] = `"hash entities do not support the \\"number\\" outputType!"`;
